core/src/main/java/org/mule/lifecycle/RegistryBrokerLifecycleManager.java

```diff
@@ -32,13 +32,12 @@ public RegistryBrokerLifecycleManager(String id, Registry object, MuleContext mu
     @Override
     protected void registerPhases()
     {
-        RegistryLifecycleCallback callback = new RegistryLifecycleCallback();
-        registerPhase(Initialisable.PHASE_NAME, new MuleContextInitialisePhase(),
-            new EmptyLifecycleCallback<AbstractRegistryBroker>());
+        RegistryLifecycleCallback callback = new RegistryLifecycleCallback(this);
+        LifecycleCallback<AbstractRegistryBroker> emptyCallback = new EmptyLifecycleCallback<>();
+        registerPhase(Initialisable.PHASE_NAME, new MuleContextInitialisePhase(), emptyCallback);
         registerPhase(Startable.PHASE_NAME, new MuleContextStartPhase(), callback);
         registerPhase(Stoppable.PHASE_NAME, new MuleContextStopPhase(), callback);
-        registerPhase(Disposable.PHASE_NAME, new MuleContextDisposePhase(),
-            new EmptyLifecycleCallback<AbstractRegistryBroker>());
+        registerPhase(Disposable.PHASE_NAME, new MuleContextDisposePhase(), emptyCallback);
```

Comment:
it's not an actual problem... it's just an ugly part of the implementation. Actually... this class should be deprecated since the RegistryBroker is. The actual "fix" would be to get rid of this unnecessary LifecycleManager when we finally wipe all the deprecated classes